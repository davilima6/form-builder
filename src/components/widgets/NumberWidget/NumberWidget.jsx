// @flow
import * as React from 'react';
import { Error as TypeError, normalize } from '../../../utils';
import {
  Label, StyledError, StyledInput, StyledLabel, WidgetWrapper,
} from '../..';

const { useCallback } = React;

type Props = {
  id: string,
  label: string,
  value?: number,
  errors?: Array<TypeError>,
  onChangeField: Function,
};

const NumberWidget = ({
  id, label, value, errors, onChangeField,
}: Props) => {
  const fieldId = `field-${id}`;
  const onChange = useCallback(({ target }: SyntheticEvent<>) => onChangeField(id, target.value), [
    id,
    onChangeField,
  ]);

  return (
    <WidgetWrapper hasError={errors.length > 0}>
      <Label htmlFor={fieldId}>
        <StyledLabel>{label}</StyledLabel>
        <StyledInput type="number" id={fieldId} name={id} value={value} onChange={onChange} />
      </Label>
      {errors.map(message => (
        <StyledError key={normalize(message)}>{message}</StyledError>
      ))}
    </WidgetWrapper>
  );
};

NumberWidget.defaultProps = {
  value: '',
  errors: [],
};

export default NumberWidget;
