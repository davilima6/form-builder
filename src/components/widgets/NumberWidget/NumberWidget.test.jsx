import React from 'react';
import { cleanup, render } from 'react-testing-library';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import NumberWidget from './NumberWidget';

describe('NumberWidget', () => {
  let component;

  beforeEach(() => {
    const props = {
      label: 'Age',
      type: 'number',
      data: 28,
      errors: [],
      onChange: () => {},
    };

    cleanup();
    component = render(<NumberWidget {...props} />);
  });

  test('renders without crashing', () => expect(component).toBeDefined());

  test('matches snapshot', () => {
    const { asFragment } = component;

    expect(asFragment()).toMatchSnapshot();
    expect(asFragment()).toHaveStyleRule('background', '#fff');
  });
});
