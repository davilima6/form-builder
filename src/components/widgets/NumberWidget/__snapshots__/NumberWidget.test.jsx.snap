// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberWidget matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
}

.c1 .label {
  font-variant: small-caps;
}

<div
    class="c0"
  >
    <label
      class="c1"
      for="field-undefined"
    >
      <span
        class="label"
      >
        Age
      </span>
      <input
        id="field-undefined"
        type="datetime-local"
        value="28"
      />
    </label>
  </div>
</DocumentFragment>
`;
